nrfutil/sandbox.ts

```diff
@@ -276,47 +324,201 @@ export class NrfutilSandbox {
+    ) => {
+        const info: Result[] = [];
+        const taskEnd: TaskEnd<Result>[] = [];
+        let stdErr: string | undefined;
+        let pid: number | undefined;
+
+        try {
+            await this.execCommand(
+                command,
+                args,
+                (data, processId) =>
+                    commonParser<Result>(
+                        data,
+                        {
+                            onProgress,
+                            onTaskBegin,
+                            onTaskEnd: end => {
+                                taskEnd.push(end);
+                                onTaskEnd?.(end);
+                            },
+                            onInfo: i => {
+                                info.push(i);
+                            },
+                            onLogging: logging => {
+                                this.onLoggingHandlers.forEach(onLogging => {
+                                    onLogging(logging, processId);
+                                });
+                            },
+                        },
+                        processId
+                    ),
+                (data, processId) => {
+                    pid = processId;
+                    stdErr = (stdErr ?? '') + data.toString();
+                },
+                controller,
+                filterEnv
+            );
+
+            if (
+                stdErr ||
+                taskEnd.filter(end => end.result === 'fail').length > 0
+            )
+                throw new Error('Task failed.');
+
+            return { taskEnd, info };
+        } catch (e) {
+            const error = e as Error;
+
+            const addPunctuation = (str: string) =>
+                str.endsWith('.') ? str.trim() : `${str.trim()}.`;
+
+            if (stdErr) {
+                error.message += `\n${addPunctuation(stdErr)}`;
+            }
+
+            const taskEndErrorMsg = taskEnd
+                .filter(end => end.result === 'fail' && !!end.message)
+                .map(end =>
+                    end.message ? `Message: ${addPunctuation(end.message)}` : ''
+                )
+                .join('\n');
+
+            if (taskEndErrorMsg) {
+                error.message += `\n${taskEndErrorMsg}`;
+            }
+
+            error.message = error.message.replaceAll('Error: ', '');
+            usageData.sendErrorReport(
+                `${
+                    pid && this.logLevel === 'trace' ? `[PID:${pid}] ` : ''
+                }${describeError(error)}`
+            );
+            throw error;
+        }
+    };
+
+    public execCommand = (
+        command: string,
+        args: string[],
+        onData: (data: Buffer, pid?: number) => void,
         onStdError: (data: Buffer, pid?: number) => void,
-        controller?: AbortController
+        controller?: AbortController,
+        filterEnv: (env: NodeJS.ProcessEnv) => NodeJS.ProcessEnv = env => env
```

Comment:
How about renaming this and everywhere else to something like `editEnv`.
Since it allows you to do (and probably will be used for) more than just filtering the env.